### ISSUE TITLE
Add loader and validator plugin to replace existing SparqlDataValidator class

### ISSUE THREAD

**Issue by pkalita-lbl** [linkml/linkml#1556]
This is to support getting parity in the new validation framework with what exists today. The existing `SparqlDataValidator` class does a bit of both loading and validating. This should be decomposed into a new loader and validator plugin classes.

**Comment by cmungall** [comment]
Agreed! The priority for validating RDF should likely be either shex or shacl however

**Comment by pkalita-lbl** [comment]
See also: https://github.com/linkml/linkml/issues/850

**Comment by nlharris** [comment]
@deepakunni3 are you able to work on this or should we unassign you?

**Comment by deepakunni3** [comment]
Thank you for the nudge @nlharris
Its on my list of TODOs :)

**Comment by turbomam** [comment]
Related: #3380 proposes adding a `MongoDBLoader` to `linkml.validator.loaders` — same pattern (non-file data source as a Loader subclass). If both land, might be worth a shared approach to URI-based source routing in the CLI.
